### ISSUE TITLE
Issuing badges for contributors

### ISSUE THREAD

**Issue by Bamieh** [nodejs/community-committee#212]
I stumbled upon this website by the mozilla foundation [Open Badges](https://openbadges.org/get-started/). The idea is simple.
> Open Badges are verifiable, portable digital badges with embedded metadata about skills and achievements. They comply with the Open Badges Specification and are shareable across the web.

I think it would be a cool incentive to motivate contribution if nodeJS [Issue Badges](https://openbadges.org/get-started/issuing-badges/) for contributors. Badges such as (1 year contributor) (New contributor) (nodeJS wizard) etc

The badges would then appear on the user's backpack in which they can share and brag about 😄 !

![image](https://user-images.githubusercontent.com/6191849/34821698-4922fd0c-f6cd-11e7-99c9-e225d32a9319.png)

I think it would be cool to experiment with this idea. What do you think!

**Comment by bnb** [comment]
Love this idea! This has been something I've wanted to do for a while, albeit in a slightly different scope. NodeSchool has done something similar with the localized NodeSchool stickers.

There are a ton of working groups, teams, committees, and initiatives in the Node.js project. Similar to how there are localized stickers for NodeSchool, I think we could have the option for WGs, teams, committees, and initiatives to create their own logo and have them printed + shipped to members of the group. 

This has been discussed previously in the [Build WG](https://github.com/nodejs/build/issues/529), as they wanted to have their own logo. I think this approach could provide a good framework for groups across the project to build out their own branding, especially if we provide a template and guide like the one that [NodeSchool has provided](https://github.com/nodeschool/nodeschool.github.io/tree/source/images/make-a-sticker).

I also love the idea of having this for things like first PR, annual Code and Learn events, NodeTogether, and the like.

**Comment by pup** [comment]
should we discuss this in today's meeting? although i think our agenda is already quite packed

**Comment by Tiriel** [comment]
This would be awesome! I'm still waiting for my "first contribution" pin's, but I'd go with a virtual badge for now :smile:

**Comment by bnb** [comment]
Added to the agenda - hopefully we can get to it!

**Comment by rachelnicole** [comment]
I love this idea too!

**Comment by gr2m** [comment]
Hey I’d love to help with that. This might be of interest: https://pubsonline.informs.org/doi/abs/10.1287/mnsc.2016.2540

tl;dr: adding a badge to discussion pages of new Wikipedia Editors increases retention rate by over 20%, with statistically significant effects for a year.

Full disclosure: the author is very close to me :)

**Comment by Tiriel** [comment]
Since we're on that, I just got an idea.

Context: last year I was involved in a kickstarter to fund a special edition of the ElePHPant plush toy, renamed Enfys for the occasion and turned rainbow, to promote diversity. it came with sticker and and various goodies.

I'm thinking these badges may be a good occasion to do some thing along the same lines. Heck, we could even make stickers about it too, but I think that needs a separate issue.

Would people be interested in this?

**Comment by amiller-gh** [comment]
Hi all! Excited to finally help kickstart this conversation.

I've done a lot of reading today about open badges and would like to propose a way to move forward with bringing Badges to Node.js 🙂 

I see three things we need to have fleshed-out for this initiative to succeed:

 1. Long-Term Management Strategy & Processes
 2. Initial Tooling Development (possibly optional)
 3. v1.0 Badge Creation Process

I've written down some of my initial thoughts below, but I'd love to (with @bnb's blessing) add Node.js Badges to the [CommComm strategic initiatives list](https://github.com/nodejs/community-committee/blob/master/STRATEGIC-INITIATIVES.md), break the below topics out into their own issues in a new `@nodejs/badges` repo for more detailed discussion, and, if there is enough interest, get a meeting on the calendar for those interested in helping to push this forward. Sound like a plan? 👍 

---

## Management Strategy
Node.js Badges will be a strategic initiative under the Community Committee with a dedicated repository to manage incoming requests and team of stakeholders who conduct regular meetings on the initiative's behalf.

I see five (5) major processes that must be established for Node.js Open Badges to succeed:

### Set Up a Badge Issuer Account
Many badge issuer services are built-in to educational tools (ex: Moodle, BadgeList, ForAllRubrics, TopClass, etc...), others are more enterprise focused (ex: Acclaim, NOCTI, Bestr, OpenBadgeFactory, Credly) 

The ideal solution for Node.js appears to be the open source service called [Badgr](https://badgr.io). Badgr appears to be the largest public player in the Open Badges space and provides a fully featured free service, with multiple user management of issuer accounts and an open API. By using Badgr for our issuer account management, we also are supporting one of the only open source implementations of an Open Badges server! 
 
### Badge / Pathway RFC process and template
We will need a process to evaluate and accept new badge and pathway proposals. I recommend we use an RFC-like process for new additions. Every new badge or pathway proposal will be submitted as a PR to add a new file to the @nodejs/open-badges repo by filling out a template markdown file with the required information. Discussion will happen on the PR and when consensus is reached by the Open Badges Team, the PR will be merged

Outstanding questions:

 - What information do we need in the RFC template?
 - Do we require a final design before merging the PR? If no, what is the design process?
 - Does the larger Community Committee want final say on badge approval?
 - What is the process for modifying badge designs / criteria? 
 - Is there a better format for creating new Badges?

### Badge Release Process
Once a new badge RFC is accepted, there will be work required outside of GitHub to officially release it. This process should be a well-defined, clone-able ticket, with manageable sub-tasks, that we can open as an issue for every new Badge.

### Badges as Stickers
Devs love their laptop stickers, and it would be amazing to offer these badges as vinyl stickers to recipients! At the end of the day, Open Badges are just images, so it will be impossible to prevent counterfeits. However, a happy-path to receiving stickers for those what want them will help to solidify Node.js Badges as an important part of the community culture.

The easiest way to make this happen is to just make all Badges available as stickers in the Node.js store. This can either be through the existing [store.nodejs.org](https://store.nodejs.org/) storefront, or through a new account from a service like [RedBubble](https://www.redbubble.com). Badge descriptions can include a link to the official store page where the badge stickers (or pins?!) may be purchased. 

It would be nice to limit official purchases of stickers to those that can prove they've been awarded the badge. I'd be interested to hear proposals for how we can pull this off, but I don't want to see it to block us from offering Badge specific merchandise.

There are also opportunities to provide Badge related merchandise at conferences and in-person meetups. This should be considered on a by-event basis, but it may be good to offer guidance for how to plan ahead for event-specific Badges.

### Badge Awards
We should try to automate badge awarding as much as possible (detailed below), but there will be certain badges that require manual approval (Ex: convention attendance, mentorship activity, meet-up hosts/volunteers). We need a process for how to submit candidates, who has access to approve, and on what cadence we process applications. All manual approvals should be judged by criteria outlined in the Badge RFCs.

## Badge Tooling Development
I think we can all agree that it would be ideal to automate the badge issuance process as much as possible. A large number Badge criteria fulfillment hooks can probably be triggered by GitHub events (ex: Addition to a GitHub team, first commit, n-number of engagements with a repo, etc...). If we integrate with an issuer platform that exposes an open API (like [Badgr](https://api.badgr.io/docs/v2/)) we can use (or create) a GitHub integration that awards badges automatically in response to GitHub events. New Badges that are issued in response to GitHub events should have to specify the exact hooks they respond to in their RFC.

In my many, many minutes-long Google-exploration, I did not find an Open Badges integration available for GitHub. The majority use case for Open Badges appear to be for formal and informal educational programs, not for open source contribution and engagement — despite it being a _perfect_ tool for OSS community development! 

If there is in fact no pre-existing way to automagically award badges based on GitHub actions, this would be an amazing contribution that Node.js can give back to the larger open source ecosystem, while we also scratch our own itch to ease long-term maintenance of Node.js Badges.

## V1 Badge Creation
We will need to determine what content will constitute our 1.0 launch of Node.js Badges. Hopefully, these Badges will follow whatever RFC-like process is agreed upon above, but by virtue of being the first batch of badges we will have some one-time hurdles to jump through. Namely: what should official Node.js Badges look like?

I would *love* to see a Node.js Badges illustration style guide emerge from this badge creation process. The style guide should provide enough of a framework to maintain visual consistency across the entire Badge collection, while providing enough flexibility to encourage creativity and variety in future Badge designs.

Examples of [Illustration style guides](https://medium.com/@alpower81/creating-an-illustration-guideline-57556a56dee2): [Atlassian](https://atlassian.design/guidelines/marketing/foundations/illustrations), [Behance](https://www.behance.net/gallery/46062029/Illustration-Guideline), [Dropbox](https://medium.com/dropbox-design/illustrating-a-more-human-brand-part-1-e1cfe3c28d9)

Note: It may be possible to extend this illustration style guide to the Node.js project as a whole – a great crossover opportunity with the Website Redesign initiative @oe !

Because we need to develop this style guide, the first batch of Badges will go through a different design process than all subsequent badge designs. We will likely need to:

 1. Determine badges we want available at "launch". We should dogfood the RFC process mentioned above for this.
 2. Use the initial badges list to develop an [Illustration Guideline](https://medium.com/@alpower81/creating-an-illustration-guideline-57556a56dee2) for Node.js Badges. Possible avenues for this: hired design agency, design competition, recruit designer friends, etc.
 3. Finalize V1 badges and roll out. Future Badges RFCs will be approved by the Badges team only if they meet style guide standards, or are approved as a special-case.

**Comment by mhdawson** [comment]
Looks like a good overview of what we need to do/discuss.  I remember the build wg discussion around a sticker and this would have would have made that more likely to happen.  One thing to include in the issues to address is how we get the ok to include the Node.js logo or derivative in the badges.  We may need a foundation review step.

**Comment by codeekage** [comment]
Nice One!!

**Comment by amiller-gh** [comment]
Closing this issue because the @nodejs/badges working group is off and running 🙂Check over in that repo for initiative progress.
